src/test/spec/unified_runner/matcher.rs

```diff
@@ -39,6 +46,72 @@ pub(crate) fn events_match(
     }
 }
 
+#[cfg(feature = "tracing-unstable")]
+pub(crate) fn tracing_events_match(
+    actual: &TracingEvent,
+    expected: &ExpectedMessage,
+) -> Result<(), String> {
+    if actual.target != expected.target {
+        return Err(format!(
+            "Expected and actual tracing event components do not match: expected {}, got {}",
+            expected.target, actual.target
+        ));
+    }
+    if actual.level != expected.level {
+        return Err(format!(
+            "Expected and actual tracing event levels do not match: expected {}. got {}",
+            expected.level, actual.level
+        ));
+    }
+
+    if let Some(has_failure) = expected.has_failure {
```

Comment:
this logic is likely going to change based on the thread I link above: re error redaction. 

potentially, we'll end up with both `has_failure` (which would simply check whether there is a key named "failure") and then `failure_is_redacted` which would assert that the error has been redacted. I wrote a sort of quick and dirty implementation of that here which checks for "REDACTED" but if this ends up being the path forward I think we could maybe match the error string against a regex too to confirm that the message is the redacted part.

Reply:
this logic has been updated. note that we redact a few different error types now and not just command errors. before I go down the path of writing regexes matching the Display impl for each possible type of error that gets redacted, do we think that is worthwhile or have any other ideas for how to make these assertions a bit more robust?